website/tutorial/basic/04-a-simple-query.mdx

```diff
@@ -0,0 +1,197 @@
+---
+id: 04-a-simple-query
+title: A simple Query
+sidebar_label: A simple Query
+---
+
+In this section, you are going to implement the first GraphQL object types and fields that provides the functionality of a Hacker News
+clone for querying a feed of _links_ that were posted by other users.
+
+### Extending the schema definition
+
+Let's start by implementing a `feed` field on the `Query` root types which allows retriving a list of `Link` elements. In general, when
+adding a new feature to the GraphQL Schema, the process will look pretty similar every time:
+
+1. Extend the GraphQL schema definition (SDL) with a new _root field_ (or new _object types_ and fields, if needed)
+1. Implement corresponding _resolver functions_ for the added fields/object types
+
+This process is also referred to as the _schema-driven_ or _schema-first development_ workflow.
+
+So, let's go ahead and tackle the first step, extending the GraphQL schema definition.
+
+<Instruction>
+
+In `src/schema.ts`, update the GraphQL schema to look as follows:
+
+```typescript
+export const typeDefinitions = /* GraphQL */ `
+  type Query {
+    info: String!
+    feed: [Link!]!
+  }
+
+  type Link {
+    id: ID!
+    description: String!
+    url: String!
+  }
+`
+```
+
+</Instruction>
+
+Pretty straightforward, right? You're defining a new `Link` object type that represents the links that can be posted to Hacker
+News. Each `Link` entity has an `id`, a `description`, and a `url` field. You're then adding another field to the root `Query` type
+that allows you to retrieve a list of `Link` elements. This list is guaranteed to never be `null` (if there isn't anything, the list must be empty) and never contain any elements that are `null` - that's what the two exclamation marks are for.
+
+> If you wish to read more about GraphQL and nullability, go ahead to [GraphQL official documentation](https://graphql.org/learn/schema/#lists-and-non-null).
+
+### Implement resolver functions
+
+The next step is to implement the resolver function for the `feed` query field. In fact, one thing we haven't mentioned yet is
+that not only _root fields_, but virtually _all_ fields on the types in a GraphQL schema have resolver functions. So,
+you'll add resolver functions for the `id`, `description`, and `url` fields of the `Link` type as well.
+
+<Instruction>
+
+In `src/schema.ts`, add a new list with dummy data as well and update the `resolvers` to look as follows:
+
+```typescript
+// 1
+type Link = {
+  id: string
+  url: string
+  description: string
+}
+
+// 2
+const links: Link[] = [
+  {
+    id: 'link-0',
+    url: 'https://graphql-yoga.com',
+    description: 'The easiest way of setting up a GraphQL server',
+  },
+]
+
+export const resolvers = {
+  Query: {
+    info: () => `This is the API of a Hackernews Clone`,
+    // 3
+    feed: () => links,
+  },
+  // 4
+  Link: {
```

Comment:
we could mention that the resolve, in this scenario, can be done at the parent level since the fields are not computed but directly fetched from the object

Reply:
Not sure if I understand this correctly, but we should explain default field resolver behavior?

Reply:
added to https://github.com/dotansimha/graphql-yoga/issues/916